src/assets/theme/index.tsx

```diff
@@ -1,8 +1,56 @@
 import { createMuiTheme } from '@material-ui/core';
-
+import { css, FlattenSimpleInterpolation } from 'styled-components';
 import palette from './palette';
 import typography from './typography';
 import overrides from './overrides';
+import { COLOR_MAP } from 'common/colors';
+
+interface ThemeColors {
+  green: string;
+  greenDark: string;
+  lightBlue: string;
+}
+
+const colors: ThemeColors = {
```

Comment:
I'm not quite sure I understand the limitations of using the colors in the palette rather than defined and included in the theme as an explicit colors property.

Also, looking at our palette, I'm not sure I understand why the values are what they are:
```
primary: {
    contrastText: black,
    dark: white,
    main: white,
    light: white,
  },
  secondary: {
    contrastText: white,
    dark: '#009e53',
    main: '#00d07d',
    light: '#61ffb0',
  },
```
here, primary is just white (? that feels incorrect) and secondary is the green color. 

It feels like the green is the primary color and blue would be the secondary color.  

I mostly bring this up because I think fixing that would simplify some of the other components and add some flexibility.  I.e. for almost all of the material ui components, you could then use the color key without having to manually add `color:` in a styled component.

Reply:
The Material design system is supposed to be customized as a whole, in a manner that is consistent with your brand (see [Crane](https://material.io/design/material-studies/crane.html) and other Material design studies). For most projects on their early stages, it's hard to have a comprehensive design system right from the start, and at this point, we don't have one.

The Material UI color variables are applied to all the Material UI components, and can be used by other components as well. Right now, it would be a bit risky to change the value of a color in the theme, because we don't know the repercussions that that would have in the components we use (if we change secondary to be light blue, will that change the color on the MiniMap tabs?).

The variables are also not super readable, so having color variables with better names help us use them more consistently in our components. If at some point we decide to restyle our site, we can style at the theme level and change the values of named colors, but we would need the design system first.

Reply:
> The Material design system is supposed to be customized as a whole, in a manner that is consistent with your brand (see [Crane](https://material.io/design/material-studies/crane.html) and other Material design studies). For most projects on their early stages, it's hard to have a comprehensive design system right from the start, and at this point, we don't have one.
> 

I think Crane is a great example of how the design system really shines, but I don't think you have to be as structured and complex to take advantage of it.  We do have a primary/secondary color that we use consistently throughout the site so I think it fits in fairly nicely with what we have. 

> The Material UI color variables are applied to all the Material UI components, and can be used by other components as well. Right now, it would be a bit risky to change the value of a color in the theme, because we don't know the repercussions that that would have in the components we use (if we change secondary to be light blue, will that change the color on the MiniMap tabs?).

I might be missing something, but I don't think that it would be too much work to clean up the colors as we have them.  There are definitely some changes, but for the most part we don't rely on the default colors used from the theme because we never really set it up properly - I made a [quick attempt at it](https://github.com/covid-projections/covid-projections/compare/pablo/component-library-setup...ghop02/component-library-but-more-centralized-colors?expand=1) and while it may take a slightly closer look I don't think it's a very drastic change. 

> The variables are also not super readable, so having color variables with better names help us use them more consistently in our components. If at some point we decide to restyle our site, we can style at the theme level and change the values of named colors, but we would need the design system first.

I'm not sure I quite understand this point? Do you have a couple examples of where we'd have to step outside of the palette to choose a color? haha I think that would be helpful so I make sure I'm not overlooking any huge issues with it

Reply:
i dont know if crane is a super applicable example. its quite theme-y/template-y, whereas our frontend is comprised of a lot of customized components that all kind of use colors differently.

re: the notion of primary/secondary, i dont think we currently have a convention that fits this exactly. green is the color of our logo, the footer, and the 'low' threshold, but thats about it? likewise our use of blue is sometimes a little disorganized.

i feel like if we want to use palette (primary, secondary, etc), we should have an index file of our own making in which we define/name these colors (green, blue, etc), and then use them to set primary/etc in palette. ie. primary.main: theme.blue

Reply:
Very true re notion of primary/secondary and it's not really obvious which one you would use.  (Sidenote but it's really fun looking at some of the other color schemes on that material design page.  [Rally](https://material.io/design/material-studies/rally.html#color) and [owl](https://material.io/design/material-studies/owl.html#color) do some fun things with color.  Rally feels closest to us, where they just have a primary color and then a couple others to choose from)

I definitely agree and also I'm realizing that I'm not putting a ton of semantic weight on what primary and secondary mean - I think the `COLOR_MAP` constant is a bit chaotic right now, and I like the fact that the theme palette is constrained more than anything.

Here's an idea (that we can talk about in the meeting today):

* change `ThemeColors` to be an object of `[colorName]: PaletteColor` (PaletteColor is a [material ui type](https://material-ui.com/customization/palette/#customization).  That way, we can customize the material-ui palette with the colors we have.  For instance here's a quick snippet of it:

```ts
const colors: ThemeColors = {
  green: {
    main: '#00D474',
    dark: '#0A3D31',
    light: '...'
  },
  blue: {
    main: '#3BBCE6',
    light: '...'
  },
  orange: {
    ...
  },
   ...
};

// This would be used in the mui theme
const palette = {
  primary: colors.green,
  secondary: colors.blue,
  warning: colors.orange,
};
```

`colors` would essentially replace `COLOR_MAP`. My thinking is that giving each color a "main", "light", and "dark" will encourage the most reusability while giving flexibility.